### PR TITLE
Fix running `gen` on package with library

### PR DESCRIPTION
**Pull request by MasloMaslane**

### CODE REVIEW

File: src/sinol_make/helpers/compile.py [hunk 1]
```diff
@@ -95,11 +96,21 @@ def compile_file(file_path: str, name: str, compilers: Compilers, weak_compilati
...
     os.makedirs(compile_log_dir, exist_ok=True)
 
+    with open(os.path.join(os.getcwd(), "config.yml"), "r") as config_file:
+        config = yaml.load(config_file, Loader=yaml.FullLoader)
+
+    extra_compilation_files = [os.path.join(os.getcwd(), "prog", file)
+                               for file in config.get("extra_compilation_files", [])]
+    extra_compilation_args = []
+    if 'extra_compilation_args' in config:
+        extra_compilation_args = config['extra_compilation_args'].get(os.path.splitext(name)[1], [])
```
Comment (tonowak) [inline src/sinol_make/helpers/compile.py:106]: Assuming that `config` is a dictionary, I bet Python has a "get" or "get_with_default" function that returns the object under the given key or a default value when the key is not present. That would make the code much shorter.

Comment (MasloMaslane) [reply src/sinol_make/helpers/compile.py]: I changed it

File: src/sinol_make/helpers/compile.py [hunk 2]
```diff
@@ -95,11 +96,19 @@ def compile_file(file_path: str, name: str, compilers: Compilers, weak_compilati
     os.makedirs(executable_dir, exist_ok=True)
     os.makedirs(compile_log_dir, exist_ok=True)
 
+    with open(os.path.join(os.getcwd(), "config.yml"), "r") as config_file:
+        config = yaml.load(config_file, Loader=yaml.FullLoader)
+
+    extra_compilation_files = [os.path.join(os.getcwd(), "prog", file)
+                               for file in config.get("extra_compilation_files", [])]
+    extra_compilation_args = config.get('extra_compilation_args', {}).get(os.path.splitext(name)[1], [])
```
Comment (tonowak) [inline src/sinol_make/helpers/compile.py:104]: We can make this change also in other places in the code. What do you think?

Comment (MasloMaslane) [reply src/sinol_make/helpers/compile.py:104]: I agree that this is the better way. I changed it in other places, but couldn't find much

**Review by tonowak — APPROVED** [review]

**Pull request merged**